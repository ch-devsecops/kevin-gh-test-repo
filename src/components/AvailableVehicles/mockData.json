{
  "uid": "1ebf252b-374c-4c65-a977-5a1bf68b2031",
  "componentName": "AvailableVehicles",
  "dataSource": "{1E951413-E986-475B-BA74-D7BA568D167D}",
  "params": {},
  "fields": {
    "data": {
      "value": {
        "anchorId": {},
        "gtmTitle": {},
        "gtmCategory": {},
        "gtmModelName": {},
        "gtmTrimName": {},
        "gtmBodyStyle": {
          "value": "GTM Body Style"
        },
        "gtmInteractionType": {},
        "catalog": {
          "fields": {
            "name": "Honda",
            "models": [
              {
                "items": [
                  {
                    "name": "ODYSSEY",
                    "modelYears": [
                      {
                        "name": "2023",
                        "trims": [
                          {
                            "name": "EX-L",
                            "transmissions": [
                              {
                                "items": [
                                  {
                                    "id": "0971BB011A9A49798362579C17773E51",
                                    "name": "Automatic",
                                    "detKey": {
                                      "value": "10797-Automatic"
                                    },
                                    "detIdentifier": {
                                      "value": "10797"
                                    },
                                    "sotId": {
                                      "value": ""
                                    },
                                    "price": {
                                      "value": "36,000.00"
                                    },
                                    "trim": {
                                      "value": "EX-L"
                                    },
                                    "tooltipText": {
                                      "value": "This is a tooltip"
                                    },
                                    "image": {
                                      "url": "../../../../sample-mock-data/media/img/grey-crv.png"
                                    },
                                    "exteriorColorKey": {
                                      "value": "Grey"
                                    }
                                  }
                                ]
                              }
                            ]
                          },
                          {
                            "name": "TOURING",
                            "transmissions": [
                              {
                                "items": [
                                  {
                                    "id": "0971BB011A9A49798362579C17773E51",
                                    "name": "Automatic",
                                    "detKey": {
                                      "value": "10798-Automatic"
                                    },
                                    "detIdentifier": {
                                      "value": "10798"
                                    },
                                    "sotId": {
                                      "value": ""
                                    },
                                    "price": {
                                      "value": "42,000.00"
                                    },
                                    "trim": {
                                      "value": "Touring"
                                    },
                                    "tooltipText": {
                                      "value": "This is Touring"
                                    },
                                    "image": {
                                      "url": "../../../../sample-mock-data/media/img/blue-accord.png"
                                    },
                                    "exteriorColorKey": {
                                      "value": "blue"
                                    }
                                  }
                                ]
                              }
                            ]
                          },
                          {
                            "name": "BLACK EDITION",
                            "transmissions": [
                              {
                                "items": [
                                  {
                                    "id": "0971BB011A9A49798362579C17773E51",
                                    "name": "Automatic",
                                    "detKey": {
                                      "value": "10799-Automatic"
                                    },
                                    "detIdentifier": {
                                      "value": "10799"
                                    },
                                    "sotId": {
                                      "value": ""
                                    },
                                    "exteriorColorKey": {
                                      "value": "orange"
                                    },
                                    "price": {
                                      "value": "57,550.00"
                                    },
                                    "trim": {
                                      "value": "black-edition"
                                    },
                                    "tooltipText": {
                                      "value": "This is Black Edition"
                                    },
                                    "image": {
                                      "url": "../../../../sample-mock-data/media/img/grey-civic.png"
                                    },
                                    "exteriorColorKey": {
                                      "value": "blue"
                                    }
                                  }
                                ]
                              }
                            ]
                          },
                          {
                            "name": "EX",
                            "transmissions": [
                              {
                                "items": [
                                  {
                                    "id": "0971BB011A9A49798362579C17773E51",
                                    "name": "Automatic",
                                    "detKey": {
                                      "value": "10796-Automatic"
                                    },
                                    "detIdentifier": {
                                      "value": "10796"
                                    },
                                    "sotId": {
                                      "value": ""
                                    },
                                    "price": {
                                      "value": ""
                                    },
                                    "trim": {
                                      "value": "AWD"
                                    },
                                    "tooltipText": {
                                      "value": "This is EX model"
                                    },
                                    "image": {
                                      "url": "../../../../sample-mock-data/media/img/orange-civic.png"
                                    },
                                    "exteriorColorKey": {
                                      "value": "blue"
                                    }
                                  }
                                ]
                              }
                            ]
                          }
                        ]
                      }
                    ]
                  }
                ]
              }
            ]
          }
        }
      }
    },
    "gtmCategory": {
      "id": "86f35331-a03f-4d3e-b8d2-6142e38ad8f3",
      "url": "/sitecore/System/Foundation/common/GTM-Categories/Others",
      "name": "Others",
      "displayName": "Others",
      "fields": {
        "value": {
          "value": "others"
        }
      }
    },
    "gtmTitle": {
      "value": "GTM Title "
    },
    "gtmBodyStyle": {
      "value": "GTM Body Style"
    },
    "gtmInteractionType": {
      "value": "GTM Interaction Type"
    },
    "gtmTrimName": {
      "value": "GTM Trim Name"
    },
    "gtmModelName": {
      "value": "GTM Model Name"
    }
  }
}
