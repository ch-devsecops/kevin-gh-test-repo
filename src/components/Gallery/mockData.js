export const strings = {
  Shared: {
    Gallery: {
      galleryModalAria: 'gallery modal',
      exitModalAria: 'exit modal',
      fullscreenModeAria: 'view full screen mode',
      nextImageAria: 'view next image',
      prevImageAria: 'view previous image',
      nextThumbnailPageAria: 'view next thumbnail page',
      prevThumbnailPageAria: 'view previous thumbnail page',
      hideAria: 'hide content',
      hideLabel: 'Hide',
    },
  },
};

export default { strings };
