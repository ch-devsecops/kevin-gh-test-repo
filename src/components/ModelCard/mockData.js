// eslint-disable-next-line import/prefer-default-export
export const ModelCardData = {
  Model: {
    Name: 'Model_Name',
    NameBadge: {
      src: '/data/media/img/model-card/ilx_badge.svg',
      alt: 'model name badge alt text',
    },
    ModelKey: 'mdx',
    modelYear: 2022,
  },
  Trims: {
    SpecialTrimType: 'Available',
    TaglineOverride: 'Performance Luxury Sedan',
    NameBadge: {
      src: '/data/media/img/model-card/type_s.svg',
      alt: 'trim name badge alt text',
    },
  },
  Transmission: {
    IsBuildable: true,
    Name: 'Transmission A',
    TransmissionModelCode: '10201-Automatic',
    HasSpecs: true,
    InteriorColorKey: 'kkebony_leatherette_^my20_ilx',
    ExteriorColorKey: 'bkebony_leatherette_^my20_rdx',
    Trim: 'rdx_10226',
  },
};
