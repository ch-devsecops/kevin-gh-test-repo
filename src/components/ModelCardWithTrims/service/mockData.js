export const Model = {
  NameBadge: '/data/media/img/model-card-with-trims/badge.svg',
  Year: {
    Tagline: 'Compact Sport Sedan',
  },
  ModelKey: 'rdx',
  ModelYear: 2020,
  Trims: [
    {
      TrimName: 'trim name',
      TrimKey: 'rdx_10226',
      TrimThumbnails: {
        PrimaryThumbnail: {
          src: '/data/media/img/model-card-with-trims/primary-thumbnail.png',
          alt: 'Primary thumbnail',
        },
      },
      NameBadge: '/data/media/img/model-card-with-trims/trim-badge.svg',
      DefaultTransmision: '10201-Automatic',
      Transmissions: [
        {
          Name: 'Transmission A',
          TransmissionModelCode: '10201-Automatic',
          IsBuildable: true,
          HasSpecs: true,
          InteriorColorKey: 'kkebony_leatherette_^my20_ilx',
          ExteriorColorKey: 'bkebony_leatherette_^my20_rdx',
        },
        {
          Name: 'Transmission B',
          TransmissionModelCode: 'transmission-b',
          IsBuildable: true,
          HasSpecs: false,
        },
        {
          Name: 'Transmission C',
          TransmissionModelCode: 'transmission-c',
          IsBuildable: false,
          HasSpecs: true,
        },
        {
          Name: 'Transmission D',
          TransmissionModelCode: 'transmission-d',
          IsBuildable: false,
          HasSpecs: false,
        },
      ],
    },
  ],
};

export const strings = {
  Shared: {
    ModelCardWithTrims: {
      exploreModelLabel: 'Explore Model',
      msrpStartingFromLabel: 'MSRP Starting From*',
      offersAvailableLabel: 'Offers Available',
      buildAndPriceLabel: 'Build & Price',
      viewTrimSpecsLabel: 'View Trim Specs',
      priceErrorLabel: 'Unable to load pricing. Please reload the page',
    },
  },
};

export default { Model, strings };
