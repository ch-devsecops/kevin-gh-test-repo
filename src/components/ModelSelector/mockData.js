export const modelSelector = {
  Title: 'Choose Your Model',
  BodyText:
    'Lorem ipsum dolor site amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor site amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor site amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
  Models: [
    {
      NameBadge: '/data/media/img/model-selector/badges/ilx.svg',
      Year: {
        Tagline: 'Compact Sport Sedan',
        Trim: {
          PrimaryThumbnail: {
            src: '/data/media/img/model-selector/models/ilx_front.png',
            alt: 'ILX front image',
          },
          SecondaryThumbnail: {
            src: '/data/media/img/model-selector/models/ilx_back.png',
            alt: 'ILX back image',
          },
        },
      },
      ModelKey: 'ilx-111',
    },
    {
      NameBadge: '/data/media/img/model-selector/badges/rlx.svg',
      Year: {
        Tagline: 'Premium Luxury Sedan',
        Trim: {
          PrimaryThumbnail: {
            src: '/data/media/img/model-selector/models/rdx_front.png',
            alt: 'RLX front image',
          },
          SecondaryThumbnail: {
            src: '/data/media/img/model-selector/models/rdx_back.png',
            alt: 'RLX back image',
          },
        },
      },
      ModelKey: 'rlx-111',
    },
    {
      NameBadge: '/data/media/img/model-selector/badges/tlx.svg',
      Year: {
        Tagline: 'Performance Luxury Sedan',
        Trim: {
          PrimaryThumbnail: {
            src: '/data/media/img/model-selector/models/tlx_front.png',
            alt: 'TLX front image',
          },
          SecondaryThumbnail: {
            src: '/data/media/img/model-selector/models/tlx_back.png',
            alt: 'TLX back image',
          },
        },
      },
      ModelKey: 'tlx-111',
    },
    {
      NameBadge: '/data/media/img/model-selector/badges/rdx.svg',
      Year: {
        Tagline: 'Luxury Crossover SUV',
        Trim: {
          PrimaryThumbnail: {
            src: '/data/media/img/model-selector/models/rdx_front.png',
            alt: 'RDX front image',
          },
          SecondaryThumbnail: {
            src: '/data/media/img/model-selector/models/rdx_back.png',
            alt: 'RDX back image',
          },
        },
      },
      ModelKey: 'rdx-111',
    },
    {
      NameBadge: '/data/media/img/model-selector/badges/mdx.svg',
      Year: {
        Tagline: 'Three-Row Luxury SUV',
        Trim: {
          PrimaryThumbnail: {
            src: '/data/media/img/model-selector/models/mdx_front.png',
            alt: 'MDX front image',
          },
          SecondaryThumbnail: {
            src: '/data/media/img/model-selector/models/mdx_back.png',
            alt: 'MDX back image',
          },
        },
      },
      ModelKey: 'mdx-111',
    },
  ],
};

export const accessoriesCtaData = {
  value: {
    href: '/accessories?key=',
    text: 'Explore Accessories',
    linktype: 'internal',
    url: '/accessories?key=',
    title: '',
    class: '',
  },
};
