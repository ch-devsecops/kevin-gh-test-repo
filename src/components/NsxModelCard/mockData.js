export const strings = {
  Shared: {
    NsxModelCard: {
      msrpStartingFromLabel: 'MSRP Starting From*',
      tooltipLabel: 'Price Tooltip Label Dictionary Value',
      priceErrorLabel: 'Unable to load pricing. Please reload the page',
    },
  },
};

export default { strings };
