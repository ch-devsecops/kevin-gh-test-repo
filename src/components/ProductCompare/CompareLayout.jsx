import React, { useMemo, useRef } from 'react';
import PropTypes from 'prop-types';
import {
  Box,
  CarouselContent,
  Column,
  Copy,
  Media as DesignSystemMedia,
  Optional,
  Row,
  useThemeContext,
} from '@honda-canada/design-system-react';

import { useProductCatalogDataTree, useProductCompareData } from './ProductCompareDataContextProvider';
import ProductSelector from './ProductSelector';
import ProductCard from './ProductCard';
import ProductSpecifications from './ProductSpecifications';
import CarouselPaginator from './CarouselPaginator';
import { useActiveProducts, useCarouselSlider } from './ProductCompareContextProvider';
import useSelectedModels from './useSelectedModels';
import useDetectStickyProductSelector from './useDetectStickyProductSelector';
import { PRODUCT_COMPARE_ID_KEY } from '../../utils/constants';
import themeStyles from './styles/ProductCompare.styles';
import { EqualHeight } from '../../utils/components/EqualHeight';
import useStickyHeightConfiguration from './useStickyHeightConfiguration';

const Media = themeStyles.apply(DesignSystemMedia, 'Media');
const ProductCompareRow = themeStyles.apply(Row, 'ProductCompareRow');
const ProductCompareColumn = themeStyles.apply(Column, 'ProductCompareColumn');
const ProductSelectorColumn = themeStyles.apply(Column, 'ProductSelectorColumn');
const NoOptionsSelected = themeStyles.apply(Copy, 'NoOptionsSelected');
const CarouselContainer = themeStyles.apply(Box, 'CarouselContainer');

const CompareLayout = ({ onRemove, onSelect, onUpdate, gtmTags, variant }) => {
  const refProductCompareRow = useRef();
  const { header } = useThemeContext();
  const productCatalogDataTree = useProductCatalogDataTree();
  const [activeProducts, { removeProduct, onProductSelected, setProducts }] = useActiveProducts();
  const selectedModels = useSelectedModels(activeProducts, productCatalogDataTree, onUpdate, setProducts);
  const { stickyHeight } = useStickyHeightConfiguration(variant, header?.desktop?.height);
  const { isSticky } = useDetectStickyProductSelector(refProductCompareRow);

  const { viewDetailsButton, selectProductsToCompare } = useProductCompareData();

  const productsSelectorList = useMemo(() => {
    const onRemoveOption = columnIndex => {
      const id = activeProducts[columnIndex];
      removeProduct(id);
      onRemove(id);
    };

    const onDropdownItemSelected = data => {
      const id = data?.[PRODUCT_COMPARE_ID_KEY];
      if (id) {
        onSelect(id);
        onProductSelected(id);
      }
    };

    const isFullProductCards = selectedModels?.every(model => !!model);

    return selectedModels?.map((currentOption, index) => {
      if (!currentOption) {
        return (
          <ProductSelectorColumn
            data-testid={`product-compare-column${index}`}
            width={[1, 1, 1 / 5]}
            key={`${index.toString()}`}
          >
            <ProductSelector
              productCatalogDataTree={productCatalogDataTree}
              selectorIndex={index}
              activeProducts={activeProducts}
              onProductSelected={onDropdownItemSelected}
              indexId={index}
            />
          </ProductSelectorColumn>
        );
      }

      return (
        <ProductCompareColumn
          width={[1, 1, 1 / 5]}
          bg={['white', 'white', index % 2 ? 'white' : 'grey.5']}
          key={`${index.toString()}-${currentOption.detIdentifier}`}
          isFullProductCards={isFullProductCards}
        >
          <ProductCard
            product={currentOption}
            onRemoveOption={onRemoveOption}
            productIndex={index}
            gtmTags={gtmTags}
            viewDetailsButton={viewDetailsButton}
            isSticky={isSticky}
            isFullProductCards={isFullProductCards}
          />
        </ProductCompareColumn>
      );
    });
  }, [activeProducts, isSticky]);

  // Carousel state and update functions
  const { prevSlide, nextSlide, setCurrentSlide, currentSlide, bindDrag, pages } = useCarouselSlider();

  return (
    <>
      <Media lessThan="desktop" stickyHeight={stickyHeight}>
        <ProductCompareRow style={{ border: 'none' }} ref={refProductCompareRow}>
          <CarouselContainer showSticky={isSticky} hideSticky={!isSticky}>
            <CarouselContent
              currentSlide={currentSlide}
              bindDrag={bindDrag}
              items={productsSelectorList}
              equalHeight
              slideAnimation
            />
            <CarouselPaginator
              pages={pages}
              currentSlide={currentSlide}
              prevSlide={prevSlide}
              nextSlide={nextSlide}
              setCurrentSlide={setCurrentSlide}
            />
          </CarouselContainer>
        </ProductCompareRow>
      </Media>
      <Media
        greaterThan="smallDesktop"
        style={{ paddingBottom: isSticky ? `${refProductCompareRow?.current?.offsetTop}px` : '0px' }}
        stickyHeight={stickyHeight}
      >
        <CarouselContainer ref={refProductCompareRow} showSticky={isSticky} hideSticky={!isSticky}>
          <EqualHeight>
            <ProductCompareRow style={{ border: 'none' }}>
              <ProductCompareColumn width={1 / 5} />
              {productsSelectorList}
            </ProductCompareRow>
          </EqualHeight>
        </CarouselContainer>
      </Media>
      <Optional when={activeProducts.length > 0}>
        <ProductSpecifications selectedModels={selectedModels} isSticky={isSticky} />
      </Optional>
      <Optional when={activeProducts.length <= 0}>
        <NoOptionsSelected
          data-testid="compare-no-selection-text"
          mt="xl"
          mb="huge"
          display="flex"
          justifyContent="center"
        >
          {selectProductsToCompare}
        </NoOptionsSelected>
      </Optional>
    </>
  );
};

CompareLayout.propTypes = {
  onRemove: PropTypes.func,
  onSelect: PropTypes.func,
  gtmTags: PropTypes.shape({
    'data-gtm-component-type': PropTypes.string,
    'data-gtm-interaction-type': PropTypes.string,
    'data-gtm-title': PropTypes.string,
  }),
  variant: PropTypes.string,
};

export default CompareLayout;
