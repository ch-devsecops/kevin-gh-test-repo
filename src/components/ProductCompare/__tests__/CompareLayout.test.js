import 'jest-styled-components';
import React from 'react';
import { axe, cleanup, jssRender } from 'test-utils';
import CompareLayout from '../CompareLayout';
import mockData from '../../../../sample-mock-data/routes/engine-compare/en.json';

afterEach(cleanup);

const engineSeries = mockData?.fields?.data?.item?.series;


describe.skip('<CompareLayout>', () => {
  const renderMock = () => jssRender(<CompareLayout catalogData={engineSeries} selectedModels={[]} />);

  it('should render without errors if there is no data', () => {
    const { container } = jssRender(<CompareLayout catalogData={{}} selectedModels={[]} />);

    expect(container).toMatchSnapshot();
  });

  it('should have no accessibility violations', async () => {
    const { container } = renderMock();

    expect(container).toMatchSnapshot();
    expect(await axe(container)).toHaveNoViolations();
  });

  it('renders without errors', async () => {
    const { container } = renderMock();

    expect(container).toMatchSnapshot();
  });
});
