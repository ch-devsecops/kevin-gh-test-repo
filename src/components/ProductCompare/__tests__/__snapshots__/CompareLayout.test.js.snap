// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompareLayout> renders without errors 1`] = `
.c1 {
  width: 100%;
  max-width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
}

.c4 {
  margin-left: auto;
  margin-right: auto;
  margin-top: m;
  display: none;
  width: 100%;
  height: 48px;
  background-color: #fff;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  bottom: 0;
  box-shadow: none;
}

.c0 {
  top: l;
  position: -webkit-sticky;
  position: sticky;
  z-index: 1;
  overflow-x: clip;
}

.c2 {
  border-bottom: 1px solid;
  border-color: #EAEAEA;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  padding-bottom: 0;
}

.c3 {
  max-width: 1280px;
  width: 100%;
  box-shadow: 0;
  background: white;
  left: 0;
  z-index: 210;
  -webkit-transition: opacity .6s ease-in;
  transition: opacity .6s ease-in;
}

.c3 > .splide > .splide__track {
  overflow: unset;
  overflow-y: visible;
}

.c5 {
  width: 20%;
  padding-left: columnGap.0;
  padding-right: columnGap.0;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  border: 1px solid;
  border-color: #EAEAEA;
  padding-left: 0;
  padding-right: 0;
}

.c6 {
  margin: 0;
  color: #2C2C2C;
  margin-top: xl;
  margin-bottom: huge;
  font-weight: normal;
  font-family: Helvetica;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  font-weight: normal;
  font: default;
  font-size: 16px;
}

@media screen and (min-width:768px) {
  .c0 {
    top: l;
  }
}

@media screen and (min-width:992px) {
  .c0 {
    top: 72px;
  }
}

@media screen and (min-width:768px) {
  .c2 {
    -webkit-flex-wrap: none;
    -ms-flex-wrap: none;
    flex-wrap: none;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    left: auto;
  }
}

@media screen and (min-width:768px) {
  .c5 {
    padding-left: columnGap.1;
    padding-right: columnGap.1;
  }
}

@media screen and (min-width:768px) {
  .c5 {
    border: 1px solid;
    border-color: #EAEAEA;
    padding-left: 0;
    padding-right: 0;
  }
}

@media screen and (min-width:992px) {
  .c5 {
    border: 0;
    padding-left: m;
    padding-right: m;
  }
}

<div>
  <div
    class="fresnel-container fresnel-lessThan-desktop c0"
  >
    <div
      class="c1 c2"
      display="flex"
      width="100%"
    >
      <div
        class="c3"
      >
        <div
          class="splide splide--slide splide--ltr splide--draggable is-active is-initialized"
          id="splide03"
        >
          <div
            class="splide__track"
            id="splide03-track"
            style="padding-left: 8px; padding-right: 8px;"
          >
            <ul
              class="splide__list"
              id="splide03-list"
              style="transform: translateX(0px);"
            />
          </div>
        </div>
        <div
          class="c4"
          display="none"
          height="48px"
          width="100%"
        />
      </div>
    </div>
  </div>
  <div
    class="fresnel-container fresnel-greaterThan-smallDesktop c0"
    style="padding-bottom: 0px;"
  >
    <div
      class=""
    >
      <div
        class="c3"
      >
        <div
          class="c1 c2"
          display="flex"
          width="100%"
        >
          <div
            class="c5"
            width="0.2"
          />
        </div>
      </div>
    </div>
  </div>
  <p
    class="sc-1k44dfv-0 c6"
    color="typographyDefault"
    data-testid="compare-no-selection-text"
    display="flex"
    font-family="default"
    font-weight="regular"
  >
    Shared.Common.selectProductsToCompareLabel
  </p>
</div>
`;

exports[`<CompareLayout> should have no accessibility violations 1`] = `
.c1 {
  width: 100%;
  max-width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
}

.c4 {
  margin-left: auto;
  margin-right: auto;
  margin-top: m;
  display: none;
  width: 100%;
  height: 48px;
  background-color: #fff;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  bottom: 0;
  box-shadow: none;
}

.c0 {
  top: l;
  position: -webkit-sticky;
  position: sticky;
  z-index: 1;
  overflow-x: clip;
}

.c2 {
  border-bottom: 1px solid;
  border-color: #EAEAEA;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  padding-bottom: 0;
}

.c3 {
  max-width: 1280px;
  width: 100%;
  box-shadow: 0;
  background: white;
  left: 0;
  z-index: 210;
  -webkit-transition: opacity .6s ease-in;
  transition: opacity .6s ease-in;
}

.c3 > .splide > .splide__track {
  overflow: unset;
  overflow-y: visible;
}

.c5 {
  width: 20%;
  padding-left: columnGap.0;
  padding-right: columnGap.0;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  border: 1px solid;
  border-color: #EAEAEA;
  padding-left: 0;
  padding-right: 0;
}

.c6 {
  margin: 0;
  color: #2C2C2C;
  margin-top: xl;
  margin-bottom: huge;
  font-weight: normal;
  font-family: Helvetica;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  font-weight: normal;
  font: default;
  font-size: 16px;
}

@media screen and (min-width:768px) {
  .c0 {
    top: l;
  }
}

@media screen and (min-width:992px) {
  .c0 {
    top: 72px;
  }
}

@media screen and (min-width:768px) {
  .c2 {
    -webkit-flex-wrap: none;
    -ms-flex-wrap: none;
    flex-wrap: none;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    left: auto;
  }
}

@media screen and (min-width:768px) {
  .c5 {
    padding-left: columnGap.1;
    padding-right: columnGap.1;
  }
}

@media screen and (min-width:768px) {
  .c5 {
    border: 1px solid;
    border-color: #EAEAEA;
    padding-left: 0;
    padding-right: 0;
  }
}

@media screen and (min-width:992px) {
  .c5 {
    border: 0;
    padding-left: m;
    padding-right: m;
  }
}

<div>
  <div
    class="fresnel-container fresnel-lessThan-desktop c0"
  >
    <div
      class="c1 c2"
      display="flex"
      width="100%"
    >
      <div
        class="c3"
      >
        <div
          class="splide splide--slide splide--ltr splide--draggable is-active is-initialized"
          id="splide02"
        >
          <div
            class="splide__track"
            id="splide02-track"
            style="padding-left: 8px; padding-right: 8px;"
          >
            <ul
              class="splide__list"
              id="splide02-list"
              style="transform: translateX(0px);"
            />
          </div>
        </div>
        <div
          class="c4"
          display="none"
          height="48px"
          width="100%"
        />
      </div>
    </div>
  </div>
  <div
    class="fresnel-container fresnel-greaterThan-smallDesktop c0"
    style="padding-bottom: 0px;"
  >
    <div
      class=""
    >
      <div
        class="c3"
      >
        <div
          class="c1 c2"
          display="flex"
          width="100%"
        >
          <div
            class="c5"
            width="0.2"
          />
        </div>
      </div>
    </div>
  </div>
  <p
    class="sc-1k44dfv-0 c6"
    color="typographyDefault"
    data-testid="compare-no-selection-text"
    display="flex"
    font-family="default"
    font-weight="regular"
  >
    Shared.Common.selectProductsToCompareLabel
  </p>
</div>
`;

exports[`<CompareLayout> should render without errors if there is no data 1`] = `
.c1 {
  width: 100%;
  max-width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
}

.c4 {
  margin-left: auto;
  margin-right: auto;
  margin-top: m;
  display: none;
  width: 100%;
  height: 48px;
  background-color: #fff;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  bottom: 0;
  box-shadow: none;
}

.c0 {
  top: l;
  position: -webkit-sticky;
  position: sticky;
  z-index: 1;
  overflow-x: clip;
}

.c2 {
  border-bottom: 1px solid;
  border-color: #EAEAEA;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  padding-bottom: 0;
}

.c3 {
  max-width: 1280px;
  width: 100%;
  box-shadow: 0;
  background: white;
  left: 0;
  z-index: 210;
  -webkit-transition: opacity .6s ease-in;
  transition: opacity .6s ease-in;
}

.c3 > .splide > .splide__track {
  overflow: unset;
  overflow-y: visible;
}

.c5 {
  width: 20%;
  padding-left: columnGap.0;
  padding-right: columnGap.0;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  border: 1px solid;
  border-color: #EAEAEA;
  padding-left: 0;
  padding-right: 0;
}

.c6 {
  margin: 0;
  color: #2C2C2C;
  margin-top: xl;
  margin-bottom: huge;
  font-weight: normal;
  font-family: Helvetica;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  font-weight: normal;
  font: default;
  font-size: 16px;
}

@media screen and (min-width:768px) {
  .c0 {
    top: l;
  }
}

@media screen and (min-width:992px) {
  .c0 {
    top: 72px;
  }
}

@media screen and (min-width:768px) {
  .c2 {
    -webkit-flex-wrap: none;
    -ms-flex-wrap: none;
    flex-wrap: none;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    left: auto;
  }
}

@media screen and (min-width:768px) {
  .c5 {
    padding-left: columnGap.1;
    padding-right: columnGap.1;
  }
}

@media screen and (min-width:768px) {
  .c5 {
    border: 1px solid;
    border-color: #EAEAEA;
    padding-left: 0;
    padding-right: 0;
  }
}

@media screen and (min-width:992px) {
  .c5 {
    border: 0;
    padding-left: m;
    padding-right: m;
  }
}

<div>
  <div
    class="fresnel-container fresnel-lessThan-desktop c0"
  >
    <div
      class="c1 c2"
      display="flex"
      width="100%"
    >
      <div
        class="c3"
      >
        <div
          class="splide splide--slide splide--ltr splide--draggable is-active is-initialized"
          id="splide01"
        >
          <div
            class="splide__track"
            id="splide01-track"
            style="padding-left: 8px; padding-right: 8px;"
          >
            <ul
              class="splide__list"
              id="splide01-list"
              style="transform: translateX(0px);"
            />
          </div>
        </div>
        <div
          class="c4"
          display="none"
          height="48px"
          width="100%"
        />
      </div>
    </div>
  </div>
  <div
    class="fresnel-container fresnel-greaterThan-smallDesktop c0"
    style="padding-bottom: 0px;"
  >
    <div
      class=""
    >
      <div
        class="c3"
      >
        <div
          class="c1 c2"
          display="flex"
          width="100%"
        >
          <div
            class="c5"
            width="0.2"
          />
        </div>
      </div>
    </div>
  </div>
  <p
    class="sc-1k44dfv-0 c6"
    color="typographyDefault"
    data-testid="compare-no-selection-text"
    display="flex"
    font-family="default"
    font-weight="regular"
  >
    Shared.Common.selectProductsToCompareLabel
  </p>
</div>
`;
